File: spec/java/src/nested/deeply/CustomFx.java
package nested.deeply;

import io.kaitai.struct.CustomDecoder;

public class CustomFx implements CustomDecoder {
    public CustomFx(int x) {
    }

    @Override
    public byte[] decode(byte[] src) {
        byte[] dst = new byte[src.length + 2];
        dst[0] = '_';
        dst[src.length + 1] = '_';
        System.arraycopy(src, 0, dst, 1, src.length);
        return dst;
    }
}

File: spec/rust/src/custom_fx.rs
extern crate kaitai;

#[allow(non_snake_case)]

pub mod Nested {
    pub mod Deeply {
        pub struct CustomFx {}

        impl CustomFx {
            pub fn new(_p_key: u8) -> Self {
                Self {}
            }
        }

        impl kaitai::CustomDecoder for CustomFx {
            fn decode(&self, bytes: &[u8]) -> Result<Vec<u8>, String> {
                let mut res = bytes.to_vec();
                res.insert(0, '_' as u8);
                res.push('_' as u8);
                Ok(res)
            }
        }
    }
}
